### DISCUSSION TITLE
Why would that code freeze ?

### DISCUSSION THREAD

**Discussion by softlion** [realm/realm-dotnet#3678] [Q&A]
It's in a unit test. But in the real app it freezes too.

```c#
        using var db = Realm.GetInstance(new InMemoryConfiguration("unittests"));

        var walletId = "someId";
        var wallet = new DbWallet { WalletId = walletId };
        await db.WriteAsync(() => { db.Add(wallet); });
```

The await  never returns, no crash occurs, it's stuck.

Any idea why ?

If I try to pause the debugger, it times out.

**Comment by peppy** [comment]
What kind of app type are you running from? It sounds like you may have a `SynchronizationContext` setup in a weird way.

**Reply by softlion** [reply]
I tried with both a unit test (xunit with rider), and in a maui app (ios and android). While debugging, it happens 100% of time. 
If not debugging, it does works in the maui app.
